libtelemetry/src/full/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -601,4 +617,23 @@ public void onResponse(Call call, Response response) throws IOException {
   private boolean isLollipopOrHigher() {
     return Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
   }
+
+  private static final class ExecutorServiceFactory {
+    private ExecutorServiceFactory() {}
+
+    private static synchronized ExecutorService create(String name, int maxSize, long keepAliveSeconds) {
+      return new ThreadPoolExecutor(0, maxSize,
+        keepAliveSeconds, TimeUnit.SECONDS, new SynchronousQueue<Runnable>(),
+        threadFactory(name));
+    }
+
+    private static ThreadFactory threadFactory(final String name) {
+      return new ThreadFactory() {
+        @Override
+        public Thread newThread(Runnable runnable) {
+          return new Thread(runnable, name);
```

Comment:
For every thread, the name will be the same, will be nice if we add a suffix.

Reply:
what would suffix be helpful for?

Reply:
So we can identify each thread and will be helpful during our debug. The default construct method of Thread class will add a suffix like this
```
public Thread() {
       init(null, null, "Thread-" + nextThreadNum(), 0);
}
```

Reply:
Still a bit vague on how thread number will help in debugging specifically in the context of this thread pool, but for the record if one of these threads crash, we will capture `tid`, `name` and its `priority`. So let's move on with this PR unless you have any other feedback.

Reply:
OK, let's move on.